### PR TITLE
feat: release automation script

### PR DESCRIPTION
**Pull request by solarw**
[#249](https://github.com/fetchai/cosmpy/issues/249)

Scripts performs generic release steps:
* checks the latest release on the github and checks current version
* if current version newer, proceed with the release
* checks history for a new version, if no history - rlease process terminated
* build packages
* push packages to pypi repo using env variables as crendetials source(username/password)
* if everything fine, make a git tag and pushes it
* make a github release for tag and using version history as a description

### CODE REVIEW

File: scripts/do-release.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import tomli
+from packaging.version import Version
+
+ROOT = Path(__file__).parent.parent
+
+
+class BaseCredentials:
```
Comment (5A11) [inline scripts/do-release.py:36]: Do we really need this parent abstract class? if no other uses, just define `EnvCredentials` straightaway.

File: scripts/do-release.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        )
+
+    def push_tag(self, current_version) -> None:
+        """Push tag to github."""
+        subprocess.check_call(f"git push origin v{current_version}", shell=True)
+
+    def make_release(self, current_version: Version, release_history: str) -> None:
+        """Make release on Github."""
+        subprocess.check_call(
+            f"""gh release create v{current_version} --title "v{current_version}" --notes "{release_history}" """,
```
Comment (5A11) [inline scripts/do-release.py:120]: Can use the GH's autogenerated changelog instead of what's in the HISTORY.md?

Comment (solarw) [reply scripts/do-release.py:120]: this requires good changelog on github, probably need to agree on merge commits format
for the moment it's better to generate manually

File: scripts/do-release.py (hunk 1 above)
Comment (solarw) [reply scripts/do-release.py]: fixed

**Pull request merged**